client/my-sites/themes/sheet.jsx

```diff
@@ -19,15 +31,89 @@ export const ThemeSheet = React.createClass( {
 		themeSlug: React.PropTypes.string,
 	},
 
+	getDefaultProps() {
+		return {
+			theme: {
+				price: '$125',
+				description_long: `
+Kitsch four loko deep v, tousled kombucha polaroid gentrify. Kitsch bushwick mixtape, ugh wayfarers artisan YOLO godard direct trade. Post-ironic YOLO helvetica, hammock small batch man bun gastropub ethical forage. Neutra retro swag, chambray polaroid deep v distillery microdosing messenger bag pabst narwhal bitters. Fingerstache retro banh mi mixtape pabst. Tote bag cred everyday carry meh ennui leggings. Austin truffaut marfa, deep v artisan kickstarter fingerstache gentrify typewriter aesthetic meditation pop-up.
+
+				Man braid meditation meggings art party occupy kale chips, raw denim aesthetic pop-up portland cred. Cold-pressed godard authentic beard offal, quinoa butcher photo booth. Literally messenger bag waistcoat cliche taxidermy, austin knausgaard freegan. Seitan master cleanse skateboard, pickled mixtape YOLO before they sold out ugh. Authentic actually ethical fanny pack squid, flannel kale chips YOLO humblebrag polaroid franzen. Pitchfork flannel mumblecore food truck. Craft beer fap 90's, heirloom shabby chic typewriter salvia listicle pabst beard tacos sustainable yuccie.`
+			}
+		};
+	},
+
+	onBackClick() {
+		page.back();
+	},
+
+	onPrimaryClick() {
+		let action;
+
+		if ( ThemeHelpers.isPremium( this.props.theme ) && ! this.props.theme.purchased && this.props.isLoggedIn ) {
+			action = purchase( this.props.theme, this.props.selectedSite, 'showcase-sheet' );
+		} else if ( this.props.theme.active ) {
+			action = customize( this.props.theme, this.props.selectedSite );
+		} else if ( this.props.isLoggedIn ) {
+			action = activate( this.props.theme, this.props.selectedSite, 'showcase-sheet' );
+		} else {
+			action = signup( this.props.theme );
+		}
+
+		this.props.dispatch( action );
+	},
+
 	render() {
+		let actionTitle;
+		if ( this.props.isLoggedIn && this.props.theme.active ) {
```

Comment:
Yeah, I think so. I held off on refactoring this, because it reminded me of the `getButtonOptions` stuff. Perhaps we could reuse some of that code?